pkg/proxy/ipvs/proxier.go

```diff
@@ -1295,12 +1305,13 @@ func (proxier *Proxier) syncProxyRules() {
 				serv.Flags |= utilipvs.FlagPersistent
 				serv.Timeout = uint32(svcInfo.StickyMaxAgeSeconds())
 			}
-			if err := proxier.syncService(svcNameString, serv, true, bindedAddresses); err == nil {
+			if appliedServ, err := proxier.syncService(svcNameString, serv, currentIPVSServices[serv.String()], true, bindedAddresses); err == nil {
 				activeIPVSServices[serv.String()] = true
 				activeBindAddrs[serv.Address.String()] = true
+				currentIPVSServices[serv.String()] = appliedServ
```

Comment:
This seems a little bit fragile. `currentIPVSServices` is suppose to represent what is on the system, mutating it based on actions performed by the controller can lead to cases where we think some IPVS service exists on the system when it doesn't. 

I'm wondering if the trade-off between performance and correctness is to call `GetVirtualServers` once at the top of the sync loop and once again at the bottom of the sync before calling `cleanLegacyService`. This way we can always asssume `currentIPVSServices` is the current state of system.

Reply:
What about doing simply not updating `currentIPVSServices` anymore? After taking a closer look at it, it seems to me that a few missing services in the map won't affect `cleanLegacyService` as it won't ever need to clean a recently added service, and in any case the next sync would make things consistent again. For now I've updated the PR removing map updates but without a new call to `GetVirtualServers`, if you think it's really necessary than I can update this again.

Reply:
Sorry for the delay here.

> What about doing simply not updating currentIPVSServices anymore? After taking a closer look at it, it seems to me that a few missing services in the map won't affect cleanLegacyService as it won't ever need to clean a recently added service

I _think_ this is safe but I need to think about this some more. It's probably true that we don't really need to care about cleaning up virtual servers we just added and even if we didn't, it's not the worse thing to clean it up on the next round.